### ISSUE TITLE
Rules for nullish coalescing and optional chaining

### ISSUE THREAD

**Issue by tao-liu-liftoff** [airbnb/javascript#2146]
I noticed that babel-preset-airbnb added the following stage 4 proposals in [v4.2.0](https://github.com/airbnb/babel-preset-airbnb/blob/master/CHANGELOG.md#420---20191114):

- @babel/plugin-proposal-nullish-coalescing-operator
- @babel/plugin-proposal-optional-chaining

Are there any rules/guidelines for these features?

**Comment by ljharb** [comment]
Once they're added to babel-preset-airbnb by airbnb folks, and once appropriate eslint rules for them exist, then they'll be added to this guide.

**Comment by braco** [comment]
Doesn't this exist now?

**Comment by ljharb** [comment]
The former has happened; I don't believe there's been any new eslint rules for them.

If there's changes needed in the configs to allow the syntax (other than #2223), happy to review a PR that adds them.

**Comment by travi** [comment]
> If there's changes needed in the configs to allow the syntax (other than #2223), happy to review a PR that adds them.

it looks like syntax support for optional-chaining was added to eslint [in v7.5.0](https://eslint.org/blog/2020/07/eslint-v7.5.0-released#eslint-v750-released), so to avoid the need to use `babel-eslint`, it looks like the `ecmaVersion` needs to be set to `2020` ([or `11`](https://eslint.org/docs/user-guide/configuring#specifying-parser-options)).

if i were to send a PR, would this be the proper place to apply that update?
https://github.com/airbnb/javascript/blob/f0df3a8680479ff0b897cd98a1eab6b156899214/packages/eslint-config-airbnb-base/rules/es6.js#L6

i dont think setting the `ecmaVersion` would require changing [peerDependencies](https://github.com/airbnb/javascript/blob/f0df3a8680479ff0b897cd98a1eab6b156899214/packages/eslint-config-airbnb-base/package.json#L68), but getting the optional-chaining support would obviously require using `v7.5.0` or newer.

**Comment by ljharb** [comment]
`es6` wouldn't be appropriate, since ES6 doesn't have optional chaining.

Which version of eslint started supporting an ecmaVersion of `2020`? https://eslint.org/blog/2019/08/eslint-v6.2.0-released implies v6.2 but doesn't explicitly mention nullish coalescing or optional chaining; https://eslint.org/blog/2020/07/eslint-v7.5.0-released however specifically talks about optional chaining.

Since we currently [allow eslint v5.16, v6.8, or v7.2](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/package.json#L68), it would indeed be a breaking change to change the ecmaVersion to 2020.

In the meantime, it seems pretty simple to set that yourself in your own config?

**Comment by travi** [comment]
> In the meantime, it seems pretty simple to set that yourself in your own config?

yep, and I've already done so: https://github.com/form8ion/eslint-config/blob/f332890c401b1d796dc459322a3f7cbc646f4d44/index.js#L11-L13, but was hoping to help here if it could be useful.

> es6 wouldn't be appropriate, since ES6 doesn't have optional chaining.

yeah, that makes sense

> Which version of eslint started supporting an ecmaVersion of 2020? eslint.org/blog/2019/08/eslint-v6.2.0-released implies v6.2

yeah, that seems like the earliest introduction, so it seems this would be premature until the lowest supported version here is raised.

sounds like it's best to continue with this just in my own config for now. thanks for all of your efforts here.
